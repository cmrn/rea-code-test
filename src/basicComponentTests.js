import React from 'react';
import { shallow } from 'enzyme';

export default function basicComponentTests(component) {
  it('allows a className to be set', () => {
    const wrapper = shallow(React.cloneElement(component, { className: "classy" }));
    expect(wrapper.props().className).toEqual(expect.stringContaining('classy'));
  });

  it('passes through other props', () => {
    const wrapper = shallow(React.cloneElement(component, { foo: "bar" }));
    expect(wrapper.props().foo).toEqual('bar');
  });
}
