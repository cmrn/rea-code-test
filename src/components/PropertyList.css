.PropertyList {
  width: 320px;
}

.PropertyList-title {
  font-size: 30px;
  font-weight: normal;
}