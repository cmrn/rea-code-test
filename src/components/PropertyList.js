import React from 'react';
import PropTypes from 'prop-types';
import Property from '../containers/Property';
import './PropertyList.css';

const PropertyList = ({propertyIds, title, hoverContent, onClick, ...props}) => (
  <div {...props}>
    <h2 className='PropertyList-title'>{title}</h2>
    {propertyIds.map(propertyId => 
      <Property 
        key={propertyId} 
        id={propertyId}
        hoverContent={hoverContent}
        onClick={onClick}
      />
    )}
  </div>
);

PropertyList.propTypes = {
  propertyIds: PropTypes.array.isRequired,
  title: PropTypes.string.isRequired,
};

export default PropertyList;
