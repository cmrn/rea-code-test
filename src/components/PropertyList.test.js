import React from 'react';
import basicComponentTests from '../basicComponentTests';
import { shallow } from 'enzyme';
import PropertyList from './PropertyList';

const dummyProps = {
  propertyIds: [],
  title: '',
};

basicComponentTests(<PropertyList {...dummyProps} />);

it('renders list title', () => {
  const wrapper = shallow(<PropertyList {...dummyProps} title='Something' />);
  expect(wrapper.find('h2').text()).toEqual('Something');
});

describe('with no properties', () => {
  it('renders no properties', () => {
    const wrapper = shallow(<PropertyList {...dummyProps} propertyIds={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('with some properties', () => {
  it('renders each property', () => {
    const wrapper = shallow(<PropertyList {...dummyProps} propertyIds={['1', '2', '3']} />);
    expect(wrapper).toMatchSnapshot();
  });
});