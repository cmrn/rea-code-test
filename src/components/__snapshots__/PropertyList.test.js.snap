// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with no properties renders no properties 1`] = `
<div>
  <h2
    className="PropertyList-title"
  />
</div>
`;

exports[`with some properties renders each property 1`] = `
<div>
  <h2
    className="PropertyList-title"
  />
  <PropertyContainer
    id="1"
    key="1"
  />
  <PropertyContainer
    id="2"
    key="2"
  />
  <PropertyContainer
    id="3"
    key="3"
  />
</div>
`;
